File: staoo-ui/src/services/flow/formTemplateService.ts
import { request } from '../../utils/request'

// 表单模板列表请求参数
export interface FormTemplateListParams {
  pageNum?: number
  pageSize?: number
  formName?: string
  formKey?: string
  status?: string
  tenantId: number
}

// 表单模板数据结构
export interface FormTemplate {
  id?: number
  formName: string
  formKey: string
  description?: string
  formConfig: string
  status?: string
  version?: number
  tenantId?: number
  createTime?: string
  createBy?: string
  updateTime?: string
  updateBy?: string
}

// 获取表单模板列表
export const getFormTemplateList = (params: FormTemplateListParams) => {
  return request.get('/api/flow/formTemplate/list', { params })
}

// 根据ID获取表单模板详情
export const getFormTemplateById = (id: number) => {
  return request.get(`/api/flow/formTemplate/getById/${id}`)
}

// 根据表单标识获取表单模板详情
export const getFormTemplateByKey = (formKey: string) => {
  return request.get(`/api/flow/formTemplate/getByKey/${formKey}`)
}

// 保存表单模板
export const saveFormTemplate = (data: FormTemplate) => {
  return request.post('/api/flow/formTemplate/save', data)
}

// 更新表单模板
export const updateFormTemplate = (data: FormTemplate & { id: number }) => {
  return request.put('/api/flow/formTemplate/update', data)
}

// 删除表单模板
export const deleteFormTemplate = (id: number) => {
  return request.delete(`/api/flow/formTemplate/delete/${id}`)
}

// 批量删除表单模板
export const batchDeleteFormTemplate = (ids: number[]) => {
  return request.delete('/api/flow/formTemplate/batchDelete', { data: ids })
}

// 发布表单模板
export const publishFormTemplate = (id: number) => {
  return request.post(`/api/flow/formTemplate/publish/${id}`)
}

// 复制表单模板
export const copyFormTemplate = (id: number) => {
  return request.post(`/api/flow/formTemplate/copy/${id}`)
}

File: staoo-flow/src/main/java/com/staoo/flow/service/FormTemplateService.java
package com.staoo.flow.service;

import com.staoo.flow.domain.FormTemplate;

import java.util.List;

/**
 * 表单模板服务接口
 * 提供表单模板相关的业务逻辑处理方法
 */
public interface FormTemplateService {
    /**
     * 根据ID查询表单模板
     * @param id 模板ID
     * @return 表单模板信息
     */
    FormTemplate getById(Long id);

    /**
     * 根据表单标识查询表单模板
     * @param formKey 表单唯一标识
     * @param tenantId 租户ID
     * @return 表单模板信息
     */
    FormTemplate getByFormKey(String formKey, Long tenantId);

    /**
     * 查询表单模板列表
     * @param formTemplate 查询条件
     * @return 表单模板列表
     */
    List<FormTemplate> getList(FormTemplate formTemplate);

    /**
     * 新增表单模板
     * @param formTemplate 表单模板信息
     * @return 影响行数
     */
    int save(FormTemplate formTemplate);

    /**
     * 更新表单模板
     * @param formTemplate 表单模板信息
     * @return 影响行数
     */
    int update(FormTemplate formTemplate);

    /**
     * 删除表单模板
     * @param id 模板ID
     * @return 影响行数
     */
    int deleteById(Long id);

    /**
     * 批量删除表单模板
     * @param ids 模板ID列表
     * @return 影响行数
     */
    int deleteByIds(Long[] ids);

    /**
     * 根据租户ID查询所有表单模板
     * @param tenantId 租户ID
     * @return 表单模板列表
     */
    List<FormTemplate> getListByTenantId(Long tenantId);

    /**
     * 发布表单模板
     * @param id 模板ID
     * @return 影响行数
     */
    int publishFormTemplate(Long id);

    /**
     * 复制表单模板
     * @param id 模板ID
     * @param newName 新模板名称
     * @return 新模板ID
     */
    Long copyFormTemplate(Long id, String newName);

    /**
     * 验证表单标识唯一性
     * @param formKey 表单标识
     * @param tenantId 租户ID
     * @param id 模板ID（更新时使用，排除自身）
     * @return 是否唯一
     */
    boolean checkFormKeyUnique(String formKey, Long tenantId, Long id);
}